### PR TITLE
Update solana pay content

### PR DESCRIPTION
**Pull request by DogukanGun**
### Problem

There was a code snippet that no longer works and contains some grammatical errors

### Summary of Changes

The code snippet has been updated, and the grammatical errors have been corrected. Also, [a pr](https://github.com/Unboxed-Software/solana-scavenger-hunt-app/pull/1) has been opened for the [related repository](https://github.com/Unboxed-Software/solana-scavenger-hunt-app)

### CODE REVIEW

**Comment by mikemaccana** [comment]
Thanks @DogukanGun ! Just checking - is this for the superteam lesson update bounty?

**Comment by DogukanGun** [comment]
> Thanks @DogukanGun ! Just checking - is this for the superteam lesson update bounty?

You're welcome! Yes, this is for the Superteam lesson update bounty.

File: content/courses/solana-pay/solana-pay.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ title: Solana Pay
 objectives:
   - Use the Solana Pay specification to build payment requests and initiate
     transactions using URLs encoded as QR codes
-  - Use the `@solana/pay` library to help with the creation of Solana Pay
-    transaction requests
+  - Use the `@solana/pay` library to assist in creating Solana Pay transaction
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:6]:
```suggestion
  - Use the `@solana/pay` library to create Solana Pay transaction
```

File: content/courses/solana-pay/solana-pay.md [hunk 2]
```diff
@@ -223,7 +223,7 @@ function get(res: NextApiResponse) {
     icon: "https://solana.com/src/img/branding/solanaLogoMark.svg",
   });
 }
-async function post(req: PublicKey, res: PublicKey) {
+async function post(req: NextApiRequest, res: NextApiResponse) {
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:226]: Good spot! I can't believe we had these as `PublicKey`.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Thanks @DogukanGun ! 

Can you please check out the rest of the file:
 - The TS needs to be formatted per CONTRIBUTING
 - As mention in the bounty, this page has additional code at https://github.com/Unboxed-Software/solana-scavenger-hunt-app/tree/solution - ou will also need to include a PR for that
  - The old code has a lot of unnecessary `else` statements - see https://stackoverflow.com/questions/46875442/unnecessary-else-after-return-no-else-return

File: content/courses/solana-pay/solana-pay.md (hunk 1 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Fixed according to comment.

**Comment by DogukanGun** [comment]
> Thanks @DogukanGun !
> 
> Can you please check out the rest of the file:
> 
> * The TS needs to be formatted per CONTRIBUTING
> * As mentioned in the bounty, this page has additional code at https://github.com/Unboxed-Software/solana-scavenger-hunt-app - you will also need to include a PR for that
> * The old code has a lot of unnecessary `else` statements - see https://stackoverflow.com/questions/46875442/unnecessary-else-after-return-no-else-return

Thanks for the review @mikemaccana 

I updated the pr according to your comments.

File: content/courses/solana-pay/solana-pay.md [hunk 3]
```diff
@@ -502,9 +499,8 @@ export default async function handler(
     return get(res);
   } else if (req.method === "POST") {
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:500]: This else is unnecessary too.

File: content/courses/solana-pay/solana-pay.md [hunk 4]
```diff
@@ -502,9 +499,8 @@ export default async function handler(
...
     return await post(req, res);
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:501]: This else is unnecessary too.

File: content/courses/solana-pay/solana-pay.md [hunk 5]
```diff
@@ -586,9 +582,9 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
         let error = err as any
         if (error.message) {
             res.status(200).json({ transaction: "", message: error.message })
-        } else {
-            res.status(500).json({ error: "error creating transaction" })
+            return
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:585]: Run prettier over this code. See https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#jsts re: markdown and prettier.

File: content/courses/solana-pay/solana-pay.md [hunk 6]
```diff
@@ -212,9 +210,8 @@ export default async function handler(
     return get(res);
   } else if (req.method === "POST") {
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:211]: This else is unnecessary.

**Comment by mikemaccana** [comment]
I can see you fixed some of the unnecessary elses. Fix the other elses, format the code (JS uses two spaces per the prettier defaults) and add a link to the updated https://github.com/Unboxed-Software/solana-scavenger-hunt-app PR.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
See previous comment.

File: content/courses/solana-pay/solana-pay.md (hunk 6 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Fixed

File: content/courses/solana-pay/solana-pay.md (hunk 4 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Fixed

File: content/courses/solana-pay/solana-pay.md (hunk 5 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md:587]: Fixed.

Screenshot 2024-08-26 at 22 01 16

Comment (mikemaccana) [reply content/courses/solana-pay/solana-pay.md:587]: Sorry, there's something you missed in https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#jsts re: markdown and prettier. Quoting:

> Note that while prettier can format Markdown, https://github.com/prettier/prettier/issues/5378 so you'll need to format the code yourself for now. To do this quickly, make a blank deleteme.ts file, paste your code in there, and then save it (allowing prettier to do its work), and paste it back into the document.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
See new note re: prettier.

**Comment by DogukanGun** [comment]
> See new note re: prettier.

Thanks for the comment. Prettier has been run over the code snippet as mentioned in the CONTRIBUTING.md.

File: content/courses/solana-pay/solana-pay.md [hunk 7]
```diff
@@ -562,34 +560,34 @@ import { PublicKey, Transaction } from "@solana/web3.js"
...
+      message: `You've found location ${id}!`,
+    });
+  } catch (err) {
+    console.log(err);
+    let error = err as any;
+    if (error.message) {
+      res.status(200).json({ transaction: "", message: error.message });
+      return;
     }
+    res.status(500).json({ error: "error creating transaction" });
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:589]: Per https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#jsts you are allowed to assume any thrown object is an Error. What this looks like:

```suggestion
  } catch (thrownObject) {
    let error = thrownObject as Error;
    res.status(200).json({ transaction: "", message: error.message });
    return;
```

**Comment by mikemaccana** [comment]
Thanks! We're getting closer. See https://github.com/Unboxed-Software/solana-scavenger-hunt-app/pull/1#pullrequestreview-2263909682 - that repo needs an update to get rid of the old dependencies, you might just want to make it use `create-solana-dapp` instead.

File: content/courses/solana-pay/solana-pay.md (hunk 7 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Thanks for the review, updated.

**Comment by DogukanGun** [comment]
@mikemaccana should I make any other changes, or should I wait until the superteam-lesson-update is finished?

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Hi @dogukangun! Pardo nthe wait, we had a new class to test, then Rustconf then breakpoint.

I'm keen to get this merged. Just fix the issues below, use https://github.com/solana-developers/scavenger-hunt-app the repo (this is forked from your https://github.com/DogukanGun/Solana-Pay) and ensure that https://github.com/DogukanGun/Solana-Pay has all the same fixes you've made in this lesson.

File: content/courses/solana-pay/solana-pay.md [hunk 8]
```diff
@@ -562,34 +560,31 @@ import { PublicKey, Transaction } from "@solana/web3.js"
...
+      new PublicKey(account),
+      new PublicKey(reference),
+      id.toString(),
+    );
+
+    res.status(200).json({
+      transaction: transaction,
+      message: `You've found location ${id}!`,
+    });
+  } catch (err) {
```
[the remark this answers is not in the document]
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Updated.

File: content/courses/solana-pay/solana-pay.md [hunk 9]
```diff
@@ -562,34 +560,31 @@ import { PublicKey, Transaction } from "@solana/web3.js"
 ...
 
 async function post(req: NextApiRequest, res: NextApiResponse) {
```
[the remark this answers is not in the document]
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: Updated.

**Comment by DogukanGun** [comment]
Thanks for the update @mikemaccana . Everything is updated.

File: content/courses/solana-pay/solana-pay.md [hunk 10]
```diff
@@ -210,11 +208,11 @@ export default async function handler(
 ) {
   if (req.method === "GET") {
```
Comment (mikemaccana) [inline content/courses/solana-pay/solana-pay.md:209]: Please fix request and response everywhere. Use find and replace.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Please fix request and response everywhere. Use find and replace.

File: content/courses/solana-pay/solana-pay.md (hunk 10 above)
Comment (DogukanGun) [reply content/courses/solana-pay/solana-pay.md]: All of them are fixed.

**Pull request merged**